tests/src/JIT/HardwareIntrinsics/X86/Shared/GenerateTests.csx

```diff
@@ -1056,10 +1052,21 @@ private static bool isImmTemplate(string name)
 private static void ProcessInput(StreamWriter testListFile, string groupName, (string templateFileName, Dictionary<string, string> templateData) input)
 
 {
-    var testName = $"{input.templateData["Method"]}.{input.templateData["RetBaseType"]}";
-
+    var testName = $"{input.templateData["Method"]}";
     var suffix = "";
 
+    if (input.templateFileName == "SimpleUnOpConvTest.template" || input.templateFileName == "SimdScalarUnOpConvTest.template" )
```

Comment:
Is this special handling needed? It seems inconsistent with the way other intrinsics are being handled.

Reply:
Unfortunately yes. The problem is that we have a conversion from a `Vector128` with base type `double` and `float` to vector with base type `Int32` and vice versa. Current file names, class names and structure names would be overwritten if we use single type as extension of it's name.  Due to that using the names of base types of 2 different `Vector128` instances participating in conversion uniquely identifies the operation and test.

Reply:
It might be a good idea to just have a separate PR which updates the naming convention.

We worked around this with the V128 vs V256 differences by having two different test project folders, but we will probably need to just fix it sooner or later.

Another possibility is just adding a field to the tuple which caries the filename (default to existing if empty string, otherwise use what the user specified).